### PR TITLE
Add electromagnetic_properties_uri

### LINKED ISSUE #44: Put parameter look-up tables into separate files
**Is your feature request related to a problem? Please describe.**
Having multiple hugh look-up tables in one material file can make the file very large and confusing

**Describe the solution you'd like**
Give the possibility to put look-up tables of individual parameters in separate files and link them with both the path to the file and the UUID of the subsequent file. The UUID enables a cross check to verify that the actual correct file is linked.

**Describe alternatives you've considered**
An alternative would be to have something like an include inline to copy the content of the other file into the main material file. But it is unclear if this is supported by JSON.

### LINKED ISSUE #45: How do we want to structure the data of properties that are defined for certain measurement conditions?
Most measurements are only valid under certain conditions, that have to be stored alongside.
The JSON format allows to store that data in many different structures and hierarchies (any combination of arrays and structs possible).
___
### Examples based on the index of refraction:
(**Note:** BRDF lookups would be another example that could additionally require &theta;<sub>in</sub>, &theta;<sub>out</sub>, &phi;<sub>out</sub>)

#### Original proposal (special treatment of the wavelength as the lowest level of hierarchy): 
```json
"index_of_refraction": [
    {
      "temperature": 300.0,
      "n": [
        [200e-9, 0.110803374],
        [205e-9, 0.111587326]
      ],
      "k": [
        [200e-9, 1.908606137],
        [205e-9, 1.969936987]
      ]
    },
    {
      "temperature": 310.0,
      "n": [
        [200e-9, 0.110803374],
        [205e-9, 0.111587326]
      ],
      "k": [
        [200e-9, 1.908606137],
        [205e-9, 1.969936987]
      ]
    }
  ]
```
#### Singular data points with explicit mention of each key:
```json
  "index_of_refraction": [
    {
      "temperature": 300.0,
      "wave_length": 200e-9,
      "n": 0.110803374,
      "k": 1.908606137
    },
    {
      "temperature": 300.0,
      "wave_length": 205e-9,
      "n": 0.111587326,
      "k": 1.969936987
    },
    {
      "temperature": 310.0,
      "wave_length": 200e-9,
      "n": 0.110803374,
      "k": 1.908606137
    },
    {
      "temperature": 310.0,
      "wave_length": 205e-9,
      "n": 0.111587326,
      "k": 1.969936987
    }
  ]
```
#### Pure array (meaning of each entry has to derived from the schema)
```json
  "index_of_refraction": [
    [300.0, 200e-9, 0.110803374, 1.908606137],
    [300.0, 205e-9, 0.111587326, 1.969936987],
    [310.0, 200e-9, 0.110803374, 1.908606137],
    [310.0, 205e-9, 0.111587326, 1.969936987]
  ]
```

#### Treating each condition as separate hierarchy level (will probably lead to deep nesting)
```json
  "index_of_refraction": {
    "temperature": [
      [
        300,
        {
          "wavelength": [
            [
              200e-9,
              {
                "n": 0.110803374,
                "k": 1.908606137
              }
            ],
            [
              205e-9,
              {
                "n": 0.111587326,
                "k": 1.969936987
              }
            ]
          ]
        }
      ],
      [
        310,
        {
          "wavelength": [
            [
              200e-9,
              {
                "n": 0.110803374,
                "k": 1.908606137
              }
            ],
            [
              205e-9,
              {
                "n": 0.111587326,
                "k": 1.969936987
              }
            ]
          ]
        }
      ]
    ]
  }
```

___

### LINKED ISSUE #47: Add permeability to the material data
**Is your feature request related to a problem? Please describe.**
Especially radar simulation needs permittivity as a physical parameter.

**Describe the solution you'd like**
As a basis for the discussion, look at the definition in the legacy OpenMATERIAL:
https://github.com/asam-ev/OpenMATERIAL/tree/OpenMATERIAL_Legacy/glTF_extensions/OpenMaterial_permeability_data

There should be some meta data, e.g. how the parameter was measured or in what literature it was looked up, for this individual parameter.

### LINKED ISSUE #49: Add permittivity to the material data
**Is your feature request related to a problem? Please describe.**
Especially radar simulation needs permittivity as a physical parameter.

**Describe the solution you'd like**
As a basis for the discussion, look at the definition in the legacy OpenMATERIAL:
https://github.com/asam-ev/OpenMATERIAL/tree/OpenMATERIAL_Legacy/glTF_extensions/OpenMaterial_permittivity_data

There should be some meta data, e.g. how the parameter was measured or in what literature it was looked up, for this individual parameter.

### PR DESCRIPTION
**Pull request by ClemensLinnhoff**
## Describe your changes

- [x] Remove index_of_refraction from the material_example
- [x] Add electromagnetic_properties_uri to material example
- [x] Add new file for the electromagnetic properties as discussed in the meeting on 21.08.2024

## Issue ticket number and link

Closes #44
Closes #45
Closes #47 
Closes #49 

## Checklist before requesting a review

- [x] I have performed a self-review of my code/documentation.
- [x] My changes generate no new warnings during the documentation generation.

### CODE REVIEW

**Comment by ClemensLinnhoff** [comment]
@KimuraDIVP, @DavidJRitter904, @SandroReith, @drsftx73 what do you think?

**Comment by SandroReith** [comment]
I think it's good to reduce the number of data points and structure the data better.  From a radar point of view, permitivity and permeabilty are also fine.

**Comment by ClemensLinnhoff** [comment]
Change file extension from xomlt to xompt to keep the p for property in there.

File: examples/example_material.xomp [hunk 1]
```diff
@@ -15,588 +15,6 @@
...
-          [1675e-9,15.34740513],
-          [1680e-9,15.38951869],
-          [1685e-9,15.4724516],
-          [1690e-9,15.49621661],
-          [1695e-9,15.53351552],
-          [1700e-9,15.55632073]
-        ]
-      }
-    ]
+    "electromagnetic_properties_uri": "example_material_emp.xomlt"
```
Comment (Nicolas-Beaugerard) [inline examples/example_material.xomp:18]: electromagnetic_properties_uri don't you meant url ?

Comment (ClemensLinnhoff) [reply examples/example_material.xomp:18]: URI stands for Uniform Resource Identifier. In this case it is a relative path to the property table file.

File: schemas/material_schema.json [hunk 2]
```diff
@@ -82,66 +82,15 @@
...
-            },
-            "required": [
-              "temperature",
-              "n",
-              "k"
-            ]
-          }
+        "electromagnetic_properties_uri": {
+          "type": "string",
+          "description": "Link to a look-up table file with electromagnetic material properties.",
```
Comment (ClemensLinnhoff) [inline schemas/material_schema.json:87]:
```suggestion
          "description": "Relative path to a property lookup table file with electromagnetic material properties.",
```

**Comment by ClemensLinnhoff** [comment]
I added all discussed changes. @drsftx73 @KimuraDIVP @AsamDiegoSanchez please formally approve so we can merge.

**Review by KimuraDIVP — APPROVED** [review]

**Review by drsftx73 — APPROVED** [review]

**Review by AsamDiegoSanchez — APPROVED** [review]
Reviewed
Signed-off-by: Diego Sánchez <diego.sanchez@asam.net>

**Pull request merged**